### ISSUE TITLE
False positive of react/prop-types with destructuring

### ISSUE THREAD

**Issue by idan** [yannickcr/eslint-plugin-react#445]
See https://gist.github.com/idan/1d31a706d5bce36d1ada#file-index-jsx-L45

The relevant bit is a handler method, which errors like so:

`45:18  error  'target' is missing in props validation  react/prop-types`

```
                 complains here
                 ↓
  handleChange ({target: {value}}) {
    this.setState({filterString: value})
    this.emitFiltered(value.trim())
  }
```

Not quite sure what additional information would be useful for running this down — let me know how I can help and I'll do so gladly.

Thanks for eslint-plugin-react! :heart:

**Comment by yannickcr** [comment]
Do you use `eslint-plugin-react@4.0.0-rc.0` ? If yes I think this is duplicate of #443.

**Comment by idan** [comment]
@yannickcr herp derp, indeed it is! Closing as duplicate.
